ea.go

```diff
@@ -28,6 +35,49 @@ type ExtendedAttribute struct {
 	Flags uint8
 }
 
+// GetFileEA retrieves the extended attributes for the file represented by `handle`. The
+// `handle` must have been opened with file access flag FILE_READ_EA (0x8).
+func GetFileEA(handle windows.Handle) ([]ExtendedAttribute, error) {
+	// default buffer size to start with
+	bufLen := 1024
+	buf := make([]byte, bufLen)
+	var iosb ioStatusBlock
+	// keep increasing the buffer size until it is large enough
+	for {
+		status := getFileEA(handle, &iosb, &buf[0], uint32(bufLen), false, 0, 0, nil, true)
```

Comment:
The ntstatus type (that's returned from getFileEA) has an Err() method that does the status and converting to dos error inside https://github.com/microsoft/go-winio/blob/87532d1cfecaa30ec76bc0b281875615ffb20a8f/pipe.go#L61